packages/marko/src/runtime/components/index.js

```diff
@@ -10,11 +10,16 @@ var FLAG_HAS_RENDER_BODY = 2;
 var FLAG_IS_LEGACY = 4;
 var FLAG_OLD_HYDRATE_NO_CREATE = 8;
 
-function safeJSONReplacer(match) {
-  if (match === "</") {
-    return "\\u003C/";
-  } else {
-    return "\\u" + match.charCodeAt(0).toString(16);
+function safeStringJSONReplacer(match) {
+  switch (match) {
+    case "'":
+      return "\\'";
+    case "\\":
+      return "\\\\";
+    case "</script":
+      return "\\u003C/script";
```

Comment:
I did some cross browser testing for spaces in the closing script and all are pretty good about it. For some reason I didn’t test spaces before the closing angle bracket. I’ll test that and update this if there is no issue there.

Reply:
I double-checked the JSON spec and [it _looks_ like `</script` can be escaped as the slightly shorter `<\\/script`](https://i.stack.imgur.com/SHLOB.gif), instead of `\\u003C/script` — it also looks like trailing garbage before the final `>` doesn't change how it closes the parent opening script tag, so that's not a concern either.

Reply:
Ah this is great! Thanks for doing some research on this one. To be clear are you saying `</script abc>` does not get parsed in any browser as the closing script?

Reply:
Unfortunately, the opposite — it gets parsed _everywhere_ according to spec as the closing script.

Reply:
Lame, okay then. We'll leave it as is, but incorporate your `<\\/script` suggestion.